### PR TITLE
fixes #13 - Allow node selection based on nodeSelector, tolerations, affinities and topologySpreadConstraints

### LINKED ISSUE #13: Allow node selection based on nodeSelector, tolerations, affinities and topologySpreadConstraints
Affinities and topologySpreadConstraints can wait for nodeSelector(enforced by validatingwebhooks) and tolerations(where all nodes run with taints) are more important to support in the helm chart

### PR DESCRIPTION
**Pull request by stzov**
This PR is addressing issue #13 in the same manner done in the cert-manager helm chart

### CODE REVIEW

**Review by prahlad141992 — APPROVED** [review]

File: deploy/charts/trust-manager/values.yaml [hunk 1]
```diff
@@ -77,6 +77,43 @@ resources: {}
...
+# for example:
+#   topologySpreadConstraints:
+#   - maxSkew: 2
+#     topologyKey: topology.kubernetes.io/zone
+#     whenUnsatisfiable: ScheduleAnyway
+#     labelSelector:
+#       matchLabels:
+#         app.kubernetes.io/instance: cert-manager
+#         app.kubernetes.io/component: controller
+topologySpreadConstraints: []
```
Comment (SgtCoDFish) [inline deploy/charts/trust-manager/values.yaml:115]: suggestion: I appreciate you've copied the descriptions from cert-manager, which is helpful. However, those links are actually broken and I raised a [cert-manager PR](https://github.com/cert-manager/cert-manager/pull/5999) to fix that. Could we update here, too?

Also, the example for `topologySpreadConstraints` mentions cert-manager and the controller; could we update that to talk about trust-manager?

**Review by SgtCoDFish — CHANGES_REQUESTED** [review]
Hey, thanks for raising this (and sorry it took me a while to get around to it!)

Really appreciate the contribution; could you take a look at my comments and see what you think?

File: deploy/charts/trust-manager/Chart.yaml [hunk 2]
```diff
@@ -17,4 +17,4 @@ sources:
 - https://github.com/cert-manager/trust-manager
 
 appVersion: v0.4.0
-version: v0.4.0
+version: v0.4.1
```
[the remark this answers is not in the document]
Comment (stzov) [reply deploy/charts/trust-manager/Chart.yaml]: Yep, I rolled back this change, I was unsure at the time about the process

File: deploy/charts/trust-manager/values.yaml (hunk 1 above)
Comment (stzov) [reply deploy/charts/trust-manager/values.yaml]: These have been updated to v1.27 and links now work

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

Thanks very much for the contribution and the rapid addressing of comments. Really appreciate it, I'll try and get this into v0.5.0 which should hopefully be released in the next couple of weeks :rocket:

**Comment by SgtCoDFish** [comment]
Ah, that failure should be simple enough to fix:

> ./hack/verify-helm-docs.sh /home/prow/go/src/github.com/cert-manager/trust-manager/bin/helm-docs
> Helm chart README.md is out of date.
> Please run 'make update-helm-docs'.

**Comment by SgtCoDFish** [comment]
> Cannot trigger testing until a trusted user reviews the PR and leaves an /ok-to-test message.

That's my bad, should've added ok-to-test straight away!

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

:rocket: Thanks very much!

**Pull request merged**